### PR TITLE
fix ignored custom callable in retrieval metric aggregation

### LINKED ISSUE #2363: Custom callable is ignored in retrieval metric aggregation
## 🐛 Bug

When passing a custom aggregation to any `RetrievalMetric`, the callable is ignored. There is a bug in the retrieval aggregation function that defaults to using the maximum of the values by default https://github.com/Lightning-AI/torchmetrics/blob/1a2f406da17a01c13ce1add83835e097460cbce3/src/torchmetrics/retrieval/base.py#L38

### To Reproduce

The following custom aggregation should output the tensor of values without aggregation but returns 1, the max of the values.

```
>>> from torchmetrics.retrieval import RetrievalNormalizedDCG
>>> indexes = tensor([0, 0, 0, 1, 1, 1, 1])
>>> preds = tensor([0.2, 0.3, 0.5, 0.1, 0.3, 0.5, 0.2])
>>> target = tensor([False, False, True, False, True, False, True])
>>> ndcg = RetrievalNormalizedDCG(aggregation=lambda x, dim: x)
>>> print(ndcg(preds, target, indexes=indexes))
tensor(1.)
>>> ndcg = RetrievalNormalizedDCG(aggregation="max")
>>> print(ndcg(preds, target, indexes=indexes))
tensor(1.)
>>> ndcg = RetrievalNormalizedDCG()
>>> print(ndcg(preds, target, indexes=indexes))
tensor(0.8467)
```

### PR DESCRIPTION
**Pull request by fschlatt**
## What does this PR do?

Fixes #2363

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
📚 Documentation preview 📚: https://torchmetrics--2364.org.readthedocs.build/en/2364/

### CODE REVIEW

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
nice, can you pls add test to cover this bug

**Comment by fschlatt** [comment]
Tests already exist. The tests compare the computed metric with a reference metric computed using sklearn. The problem is that the computed and reference metrics are aggregated using the same faulty aggregation function. I would propose to copy the fixed aggregation function into the tests. That seems redundant, though.

**Review by SkafteNicki — APPROVED** [review]
